test/integration/testdata/structtype/granular-struct.go

```diff
@@ -0,0 +1,8 @@
+package structtype
+
+// +k8s:openapi-gen=true
+type GranularStruct struct {
+	// +structType=granular
```

Comment:
is this in the right place? the structType is annotating a string. Should it fail some sort of validation?
(and same problem above on the `AtomicStruct`)

Reply:
Yeah, the annotation is applied to the string; "correctly" (in that `Field` gets the right tag in the produced API spec) but it's super confusing the way I've written it. I'll change it so that the marker annotates a "proper" struct.

Reply:
Indeed interesting that the validation didn't fail though when using `structType` on a string 🤔 I'll experiment with it and see if there's a problem here 🤔